build-scripts/build-custom-app.sh

```diff
@@ -0,0 +1,251 @@
+	echo "You are trying to duplicate a Custom App, ${new_dir} already exists. Please delete this folder and its contents and try again if you want to create a new custom app at this path"
+else
+	printf "The custom app does not exist, creating it now with the required sub-directories and files..."
+	mkdir -p "${new_dir}" "${components_dir}" "${layouts_dir}" "${pages_dir}" "${styles_dir}" "${tests_dir}" "${e2e_test_dir}"
+	printf " -> Created custom app directory for %s: %s\n\n" "${theme}" "${new_dir}"
+
+	# Create the required files
+	touch "${components_dir}/index.ts"
+	touch "${components_dir}/Analytics.tsx"
+
+	echo "${ANALYTICS_CONTENT}" >"${components_dir}/Analytics.tsx"
+	echo "${INDEX_TS_CONTENT}" >"${components_dir}/index.ts"
+	cp -r "${themes_path}/cclw/components/LandingSearchForm.tsx" "${components_dir}/LandingSearchForm.tsx"
+	printf "%s" "${HERO_CONTENT}" >"${components_dir}/Hero.tsx"
+
+	touch "${components_dir}/MethodologyLink.tsx"
+	echo "${METHODOLOGY_LINK_CONTENT}" >"${components_dir}/MethodologyLink.tsx"
+
+	touch "${pages_dir}/homepage.tsx"
+	touch "${pages_dir}/methodology.tsx"
+	echo "${HOMEPAGE_CONTENT}" >"${pages_dir}/homepage.tsx"
+	echo "${METHODOLOGY_PAGE_CONTENT}" >"${pages_dir}/methodology.tsx"
+
+	touch "${layouts_dir}/main.tsx"
+	echo "${MAIN_CONTENT}" >"${layouts_dir}/main.tsx"
+
+	# Copy the tailwind.config.ts file from the cclw theme directory
+	cp -r "${themes_path}/cclw/tailwind.config.js" "${new_dir}/tailwind.config.js"
+	printf "Copied tailwind.config.js file from %s to %s.\n\n" "${themes_path}/cclw" "${new_dir}"
+
+	# Copy the styles.scss file from the cclw theme directory
+	cp -r "${themes_path}/cclw/styles/styles.scss" "${styles_dir}/styles.scss"
+	printf "Copied styles.scss file from %s to %s.\n\n" "${themes_path}/cclw" "${styles_dir}"
+
+	# Create the redirects.json file
+	touch "${new_dir}/redirects.json"
+	echo "[]" >"${new_dir}/redirects.json"
+
+	# Create the routes.json file
+	touch "${new_dir}/routes.json"
+	echo "${ROUTES_JSON_CONTENT}" >"${new_dir}/routes.json"
+
+	# Create the e2e test file
+	e2e_test_pages_dir="${e2e_test_dir}/pages"
+	mkdir "${e2e_test_pages_dir}"
+	touch "${e2e_test_pages_dir}/homepage.cy.js"
+
+	# Add the new custom app to ci-cd.yml file
+	sed -i '' "s/theme: \[\(.*\)\]/theme: [\1, ${theme}]/" .github/workflows/ci-cd.yml
+	echo "Added ${theme} to the theme list in ci-cd.yml"
+
+	# Add the new custom app path to tsconfig.json
+	tsconfig_file="tsconfig.json"
+	temp_tsconfig="temp_tsconfig.json"
+
+	awk -v app="${theme}" '
+	BEGIN { in_paths = 0; added = 0 }
+	/"paths":/  { in_paths = 1 }
+	in_paths && /}/ && !added {
+		sub(/}/, ",\n    \"@" app "/*\": [\"themes/" app "/*\"]\n  }")
+		added = 1
+	}
+	{ print }
+	' "${tsconfig_file}" >"${temp_tsconfig}" && mv "${temp_tsconfig}" "${tsconfig_file}"
+
+	echo "Added @${theme}/* path to tsconfig.json"
+
+	# Update TTheme type in types.ts
+	sed -i.bak "s/export type TTheme = /export type TTheme = \"${theme}\" | /" src/types/types.ts
```

Comment:
This creates a file called `types.ts.bak`. What is this file used for?

Reply:
ahh it was just a side effect of the the sed command is being used on macOS, will update to remove